### PR TITLE
List: :sparkles: New size 'large'

### PR DESCRIPTION
**Pull request by HalvorHaugan**
### Description

Closes https://github.com/navikt/team-aksel/issues/641

### Component Checklist 📝

- [x] JSDoc
- [x] Examples
- [x] Documentation
- [x] Storybook
- [x] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [x] Component tokens (`@navikt/core/css/tokens.json`)
- [x] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [x] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [x] New component? CSS import (`@navikt/core/css/index.css`)
- [x] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: @navikt/core/react/src/list/List.tsx [hunk 1]
```diff
@@ -58,13 +58,19 @@ export const List = forwardRef<HTMLDivElement, ListProps>(
...
           {title && (
             <Heading
-              size={listSize === "medium" ? "small" : "xsmall"}
+              size={
+                listSize === "large"
+                  ? "medium"
+                  : listSize === "medium"
+                    ? "small"
+                    : "xsmall"
+              }
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/list/List.tsx]: What do we think about this? Could also do:
```
const headingSizeMap: Record<
  Exclude<ListProps["size"], undefined>,
  HeadingProps["size"]
> = {
  small: "xsmall",
  medium: "small",
  large: "medium",
};

(...)

<Heading size={headingSizeMap[listSize]} (...)
```

Comment (KenAJoh) [reply @navikt/core/react/src/list/List.tsx]: Since it's only one place i think the current version works _ok_ without adding any extra complexity.

**Review by KenAJoh — APPROVED** [review]
LGTM

While not related to this PR
```
    const { size: contextSize } = useContext(ListContext);
    const listSize = size ?? contextSize;
```
One can now in theory have two different sizes when nesting the list, should the "parent" list override this?

File: @navikt/core/react/src/list/List.tsx (hunk 1 above)
Comment (HalvorHaugan) [reply @navikt/core/react/src/list/List.tsx]: Oops, didn't see this comment until now 😅 I change it b.c. Julian didn't like the nested ternary, and I think I agree.

Comment (KenAJoh) [reply @navikt/core/react/src/list/List.tsx]: Agree that `headingSizeMap´ looks better 👍

**Comment by HalvorHaugan** [comment]
> LGTM
> 
> While not related to this PR
> 
> ```
>     const { size: contextSize } = useContext(ListContext);
>     const listSize = size ?? contextSize;
> ```
> 
> One can now in theory have two different sizes when nesting the list, should the "parent" list override this?

I think that's a design question, but personally I think it's nice to be able to e.g. have a nested list that is smaller.

**Pull request merged**